app/scripts/lib/crypto/a256gcm.js

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import importFxaCryptoDeriver from './deriver';
+import required from '../required';
+
+const ENCRYPTION_ALGORITHM = 'A256GCM';
+
+export default {
+
+  /**
+   * Create a JWK from `key` and `kid`. `key`
+   * is the output of hdkf (see ./hkdf.js)
+   *
+   * @param {Buffer} key
+   * @param {Object} [kid]
+   * @returns {Promise} resolves to the JWK
+   */
+  createJwkFromKey (key, kid) {
+    return importFxaCryptoDeriver().then(({jose}) => {
+      required(key, 'key');
+
+      const keyOptions = {
+        alg: ENCRYPTION_ALGORITHM,
+        k: key,
+        kty: 'oct'
+      };
+
+      if (typeof kid !== 'undefined') {
+        keyOptions.kid = kid;
+      }
+
+      return jose.JWK.asKey(keyOptions);
+    });
+  },
+
+  /**
+   * encrypts `plaintext` using `keysJwk`
+   *
+   * @param {String} plaintext Data to encrypt
+   * @param {String} keysJwk keysJwk to encrypt data
+   * @param {Object} [options={}] Options
+   *   @param {String} [options.unsafeExplicitIV] - Initialization vector used to create bundle for testing purposes
```

Comment:
Excellent name! Is there (or could we add) a lint rule that warns on `unsafeExplicit...` property names in `app/scripts`?

Reply:
@vbudhram deserves the credit, I'm not smart enough to come up with a name like that.

Reply:
@philbooth - mind checking the 2nd commit of this PR? I used grunt-ban-words to check for unsafeExplicitIV. I could do something like [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/no-danger.js) and ban just that word, though that seems like more work.

Reply:
I can't take credit either, it was inspired by rfk!